libecs/include/pivot/ecs/Core/Component/DenseComponentArray.hxx

```diff
@@ -66,17 +67,17 @@ public:
 
     /// Returns a mutable view into the components values. Some of those values can be nonsensical as the entity can
     /// miss this component.
-    std::span<T> getData() { return this->m_components; }
+    virtual std::span<T> getData() { return this->m_components; }
 
     /// Returns a constant view into the components values. Some of those values can be nonsensical as the entity can
     /// miss this component.
-    std::span<const T> getData() const { return this->m_components; }
+    virtual std::span<const T> getData() const { return this->m_components; }
 
     /// Returns the booleans specifying whether an an entity has the component
-    const std::vector<bool> &getExistence() const { return this->m_component_exist; }
+    virtual const std::vector<bool> &getExistence() const { return this->m_component_exist; }
 
     /// Returns the vector containing the component data
-    const std::vector<T> &getComponents() const { return this->m_components; }
+    virtual const std::vector<T> &getComponents() const { return this->m_components; }
```

Comment:
Why did you make all those methods virtual if now you're not using inheritance ? :thinking:

Reply:
Fixed, but virtual qualifier is removed by the compiler if the function is never overrode anywhere